consensus/src/threshold_simplex/mod.rs

```diff
@@ -3709,4 +3709,156 @@ mod tests {
+                partition: validator.to_string(),
+                mailbox_size: 64,
+                epoch: 333,
+                namespace: namespace.clone(),
+                leader_timeout: Duration::from_millis(50),
+                notarization_timeout: Duration::from_millis(100),
+                nullify_retry: Duration::from_millis(250),
+                fetch_timeout: Duration::from_millis(50),
+                activity_timeout: 4,
+                skip_timeout: 2,
+                max_fetch_count: 1,
+                fetch_rate_per_peer: Quota::per_second(NZU32!(10)),
+                fetch_concurrent: 1,
+                replay_buffer: NZUsize!(1024 * 16),
+                write_buffer: NZUsize!(1024 * 16),
+                buffer_pool: PoolRef::new(PAGE_SIZE, PAGE_CACHE_SIZE),
+            };
+            let engine = Engine::new(context.with_label("engine"), cfg);
+
+            // Start engine
+            let (pending, recovered, resolver) = registrations
+                .remove(&validator)
+                .expect("validator should be registered");
+            let handle = engine.start(pending, recovered, resolver);
+
+            // Allow tasks to start
+            context.sleep(Duration::from_millis(1000)).await;
+
+            // Verify that engine and child actors are running
+            let metrics_before = context.encode();
+            let is_running = |name: &str| -> bool {
+                metrics_before.lines().any(|line| {
+                    line.starts_with("runtime_tasks_running{")
+                        && line.contains(&format!("name=\"{}\"", name))
+                        && line.contains("task=\"Future\"")
+                        && line.trim_end().ends_with(" 1")
+                })
+            };
+            assert!(is_running("engine"));
+            assert!(is_running("engine_batcher"));
+            assert!(is_running("engine_voter"));
+            assert!(is_running("engine_resolver"));
+
+            // Abort engine and ensure children stop
+            handle.abort();
+            let _ = handle.await; // ensure parent tear-down runs
+
+            // Give the runtime a tick to process aborts
+            context.sleep(Duration::from_millis(1000)).await;
+
+            let metrics_after = context.encode();
+            let is_stopped = |name: &str| -> bool {
+                // Either the gauge is 0, or the entry is absent (both imply not running)
+                metrics_after.lines().any(|line| {
+                    line.starts_with("runtime_tasks_running{")
+                        && line.contains(&format!("name=\"{}\"", name))
+                        && line.contains("task=\"Future\"")
+                        && line.trim_end().ends_with(" 0")
+                })
+            };
+            assert!(is_stopped("engine"));
+            assert!(is_stopped("engine_batcher"));
+            assert!(is_stopped("engine_voter"));
+            assert!(is_stopped("engine_resolver"));
```

Comment:
The test panics on these lines. The `engine` is stopped but the child tasks are not

This can be confirmed by printing out the `metrics_after`